model-server/src/main/kotlin/org/modelix/model/server/handlers/ContentExplorer.kt

```diff
@@ -112,19 +98,24 @@ class ContentExplorer(private val client: IModelClient, private val repoManager:
                 )
             }
             get<Paths.getNodeIdForVersionHash> {
-                val id = call.parameters["nodeId"]!!.toLong()
-                var found: PNodeAdapter? = null
-                for (node in rootNodes) {
-                    val candidate = PNodeAdapter(id, node.branch).takeIf { it.isValid }
-                    if (candidate != null) {
-                        found = candidate
-                        break
-                    }
+                val id = call.parameters["nodeId"]?.toLongOrNull()
+                    ?: return@get call.respondText("node id not found", status = HttpStatusCode.NotFound)
+
+                val versionHash = call.parameters["versionHash"]
+                    ?: return@get call.respondText("version hash not found", status = HttpStatusCode.NotFound)
```

Comment:
Another solution to the ugly `return@` syntax would be installing the ErrorPages plugin and then mapping custom Exceptions to error responses. This would result in being able to raise specific exceptions here.